### PR TITLE
Changed behavior of JulianDate.fromIso8601 to match Javascript's Date specification

### PR DESCRIPTION
**Pull request by tfili**
In the Date spec (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date), under the description of `dateString` you'll see this note
```
Note: parsing of date strings with the Date constructor (and Date.parse, they are equivalent)
is strongly discouraged due to browser differences and inconsistencies. Support for RFC 2822
format strings is by convention only. Support for ISO 8601 formats differs in that date-only
strings (e.g. "1970-01-01") are treated as UTC, not local.
```

Also fixed/added tests and updated CHANGES.md with the breaking change.

### CODE REVIEW

File: Specs/Core/JulianDateSpec.js [hunk 1]
```diff
@@ -166,75 +166,75 @@ defineSuite([
 
     it('Construct from ISO8601 local calendar date, basic format', function() {
         var expectedDate = JulianDate.fromDate(new Date(2009, 7, 1));
-        var computedDate = JulianDate.fromIso8601('20090801');
+        var computedDate = JulianDate.fromIso8601('20090801T00');
```
Comment (mramato) [inline Specs/Core/JulianDateSpec.js]: Since these tests were originally meant to test the functionality you're changing, I would recommend removing the word local from each test and using `new Date(Date.UTC(` instead of changing the values.  Then you can remove your new tests because they will be covered by these old ones.

**Comment by mramato** [comment]
Just the one comment.  I'll always be amazed at how many specs have unspecified edge cases.

**Comment by tfili** [comment]
@mramato Should be good to go.

**Comment by mramato** [comment]
Thanks.
